src/Stryker.Core/Stryker.Core/TestRunners/DotnetTest/DotnetTestRunner.cs

```diff
@@ -14,33 +15,38 @@ public class DotnetTestRunner : ITestRunner
         private readonly OptimizationFlags _flags;
         private readonly ILogger _logger;
         private readonly string _path;
+        private readonly string _projectFile;
         private readonly IProcessExecutor _processExecutor;
-        public TestCoverageInfos CoverageMutants { get; }
 
         public DotnetTestRunner(string path, IProcessExecutor processProxy, OptimizationFlags flags, ILogger logger = null)
         {
             _logger = logger ?? ApplicationLogging.LoggerFactory.CreateLogger<DotnetTestRunner>();
 
             _flags = flags;
             _path = Path.GetDirectoryName(FilePathUtils.ConvertPathSeparators(path));
+            _projectFile = path;
             _processExecutor = processProxy;
             CoverageMutants = new TestCoverageInfos();
         }
 
-        public TestRunResult RunAll(int? timeoutMs, int? activeMutationId)
+        public TestCoverageInfos CoverageMutants { get; }
+
+        public IEnumerable<TestDescription> Tests => null;
```

Comment:
This breaks if someone uses dotnet testrunner combined with statistics reporter because TestRunner.Tests is null for dotnet testrunner and TestStatisticsReporter has no null check (and imo should not need one. It should be able to trust that it doesn't receive null data). Perhaps TestRunner.Tests in dotnet testrunner should not be null but empty list.

Reply:
Statistics reporting will be made through a dedicated PR.
I shared your opinion here. My current refactoring effort points toward store coverage data within the Mutant class to improve consistency, using special value for when tests/coverage is not known in details.

Reply:
Maybe I'm missing something but as far as I can tell you already added statistics reporting to this PR, so this will indeed break for this PR.

Reply:
Ah nevermind, didn't see previous comment yet. I agree with removing the reporting from this PR, that makes this PR much cleaner.